EVNSmartmeterMQTT_V01.py

```diff
@@ -63,40 +84,53 @@ def s8(value):
 
 
 #MQTT Init
-if useMQTT:
+if config['useMQTT']:
+    import paho.mqtt.client as mqtt
     try:
         client = mqtt.Client("SmartMeter")
-        client.username_pw_set(mqttuser, mqttpasswort)
-        client.connect(mqttBroker, mqttport)
+        client.username_pw_set(config['mqttuser'], config['mqttpasswort'])
+        client.connect(config['mqttBroker'], config['mqttport'])
     except:
         print("Die Ip Adresse des Brokers ist falsch!")
         sys.exit()
 
+# REST API
+if config['useREST']:
+    import requests
+    from requests.auth import HTTPBasicAuth
+
     
 tr = GXDLMSTranslator(TranslatorOutputType.SIMPLE_XML)
-serIn = serial.Serial( port=comport,
-         baudrate=2400,
+serIn = serial.Serial( port=config['port'],
+         baudrate=config['baudrate'],
          bytesize=serial.EIGHTBITS,
          parity=serial.PARITY_NONE,
          stopbits=serial.STOPBITS_ONE
 )
 
+if serIn.isOpen() == True:
```

Comment:
I first tried your script, since you already refactored this into the json config and you already support REST, which fits my usecase with influxdb+grafana.

Your scripts stops here for me, since my serIn is already open. 

The original scripts dont have this check and work for me. I wonder why did you introduce them? For me it only works if the port check does not exit or is removed at all.

Reply:
Hi,

you are correct, this was not helping anything. I've removed it.